data_structures/stack/dynamic_stack.c

```diff
@@ -0,0 +1,244 @@
+/**
+ * @file
+ *
+ * @details
+ * Dynamic [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)), just like Dynamic Array, is a stack data structure whose the length or capacity (maximum number of elements that can be stored) increases or decreases in real time based on the operations (like insertion or deletion) performed on it.
+ *
+ * In this implementation, functions such as PUSH, POP, PEEK, show_capacity, isempty, and stack_size are coded to implement dynamic stack.
+ *
+ * @author [SahilK-027](https://github.com/SahilK-027)
+ *
+ */
+#include <assert.h> /// to verify assumptions made by the program and print a diagnostic message if this assumption is false.
+#include <stdio.h>  /// for IO operations
+#include <stdlib.h> /// for including functions involving memory allocation such as `malloc`
+
+/**
+ * @brief DArrayStack Structure of stack.
+ */
+typedef struct DArrayStack
+{
+    int capacity, top; /// to store capacity and top of the stack
+    int *arrPtr;       ///< array pointer
+} DArrayStack;
+
+/**
+ * @brief Create a Stack object
+ *
+ * @param cap: Capacity of stack
+ * @return DArrayStack*: Newly created stack object pointer
```

Comment:
I don't think the `:` is needed in this case.
Doxygen will not find any parameter named `cap:`.

Reply:
Sure I'll remove all the `:` s. 👍🏻